File: Backend/dependency-dotnet/MHP.CodingChallenge.Backend.Dependency.Inquiry/InquiryService.cs
using System;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Logging;

namespace MHP.CodingChallenge.Backend.Dependency.Inquiry
{
    public class InquiryService
    {
        public void Create(Inquiry inquiry)
        {
            Console.WriteLine(string.Format("User sent inquiry: {0}", inquiry.ToString()));
        }
    }
}

File: inquiry/src/main/kotlin/com/mhp/coding/challenges/dependency/inquiry/InquiryService.kt
package com.mhp.coding.challenges.dependency.inquiry

import mu.KotlinLogging
import org.springframework.stereotype.Component

private val logger = KotlinLogging.logger {}

@Component
class InquiryService {
    fun create(inquiry: Inquiry) {
        logger.info {
            "User sent inquiry: $inquiry"
        }
    }
}

data class Inquiry(
    var username: String,
    var recipient: String,
    var text: String,
)
